backend/src/main/java/heartbeat/service/report/GenerateReporterService.java

```diff
@@ -21,33 +27,57 @@
 
 public class GenerateReporterService {
 
-	// todo: need remove private fields not use void function when finish GenerateReport
-	private CardCollection cardCollection;
-
 	private final JiraService jiraService;
 
+	private final BuildKiteService buildKiteService;
+
+	private final CalculateDeploymentFrequency calculateDeploymentFrequency;
+
 	// need add GitHubMetrics and BuildKiteMetrics
 	private final List<String> kanbanMetrics = Stream
 		.of(RequireDataEnum.VELOCITY, RequireDataEnum.CYCLE_TIME, RequireDataEnum.CLASSIFICATION)
 		.map(RequireDataEnum::getValue)
 		.toList();
 
+	private final List<String> buildKiteMetrics = Stream
+		.of(RequireDataEnum.CHANGE_FAILURE_RATE, RequireDataEnum.DEPLOYMENT_FREQUENCY)
+		.map(RequireDataEnum::getValue)
+		.toList();
+
+	// todo: need remove private fields not use void function when finish GenerateReport
+	private CardCollection cardCollection;
+
+	private List<DeployTimes> deployTimesList = new ArrayList<>();
+
+	private List<Map.Entry<String, List<BuildKiteBuildInfo>>> buildInfosList = new ArrayList<>();
+
 	public GenerateReportResponse generateReporter(GenerateReportRequest request) {
 		// fetch data for calculate
 		this.fetchOriginalData(request);
 
 		// calculate all required data
-		Velocity velocity = calculateVelocity();
 		calculateClassification();
-		calculateDeployment();
 		calculateCycleTime();
 		calculateLeadTime();
 
-		log.info("Successfully generate Report, request: {}, report: {}", request,
-				GenerateReportResponse.builder().velocity(velocity).build());
-
-		// combined data to GenerateReportResponse
-		return GenerateReportResponse.builder().velocity(velocity).build();
+		GenerateReportResponse reportResponse = new GenerateReportResponse();
+		request.getMetrics().forEach((metrics) -> {
+			switch (metrics.toLowerCase()) {
+				case "velocity":
+					reportResponse.setVelocity(calculateVelocity());
```

Comment:
creating a class `Velocity` to calculate is better to get the value

Reply:
same as `calculateDeploymentFrequency`

Reply:
Will be updated in yishen's PR @Eason-T